src/NzbDrone.Core/Download/Clients/Tribler/TriblerDownloadClient.cs

```diff
@@ -0,0 +1,317 @@
+        {
+            _proxy = triblerDownloadClientProxy;
+        }
+
+        public override string Name => "Tribler";
+
+        public override bool PreferTorrentFile => false;
+
+        public override IEnumerable<DownloadClientItem> GetItems()
+        {
+            var configAsync = _proxy.GetConfig(Settings);
+
+            var items = new List<DownloadClientItem>();
+
+            var downloads = _proxy.GetDownloads(Settings);
+
+            foreach (var download in downloads)
+            {
+                // If totalsize == 0 the torrent is a magnet downloading metadata
+                if (download.Size == null || download.Size == 0)
+                {
+                    continue;
+                }
+
+                var item = new DownloadClientItem
+                {
+                    DownloadId = download.Infohash,
+                    Title = download.Name,
+
+                    DownloadClientInfo = DownloadClientItemClientInfo.FromDownloadClient(this, false) // TODO: WHAT IS POST-IMPORT
+                };
+
+                // some concurrency could make this faster.
+                var files = _proxy.GetDownloadFiles(Settings, download);
+
+                item.OutputPath = new OsPath(download.Destination);
+
+                if (files.Count == 1)
+                {
+                    item.OutputPath += files.First().Name;
+                }
+                else
+                {
+                    item.OutputPath += item.Title;
+                }
+
+                item.TotalSize = (long)download.Size;
+                item.RemainingSize = (long)(download.Size * (1 - download.Progress)); // TODO: i expect progress to be between 0 and 1
+                item.SeedRatio = download.AllTimeRatio;
+
+                if (download.Eta.HasValue)
+                {
+                    if (download.Eta.Value >= TimeSpan.FromDays(365).TotalSeconds)
+                    {
+                        item.RemainingTime = TimeSpan.FromDays(365);
+                    }
+                    else if (download.Eta.Value < 0)
+                    {
+                        item.RemainingTime = TimeSpan.FromSeconds(0);
+                    }
+                    else
+                    {
+                        item.RemainingTime = TimeSpan.FromSeconds(download.Eta.Value);
+                    }
+                }
+
+                // TODO: the item's message should not be equal to Error
+                item.Message = download.Error;
+
+                // tribler always saves files unencrypted to disk.
+                item.IsEncrypted = false;
+
+                // state handling
+
+                // TODO: impossible states?
+                // Failed = 4,
+                // Warning = 5
+
+                // Queued = 0,
+                // Completed = 3,
+                // Downloading = 2,
+
+                // Paused is guesstimated
+                // Paused = 1,
```

Comment:
Anything to do here or should this be removed?

Reply:
it was mostly a development note.